### ISSUE TITLE
Add Effect Script helper functions

### ISSUE THREAD

**Issue by StevenEvans1966** [moo-man/WFRP4e-FoundryVTT#2000]
In the Effect scrips I see this a lot, some magic GUID that gets found converted to an object set a value on it then add it to an actor

`let ward = await fromUuid("Compendium.wfrp4e-core.items.Bvd2aZ0gQUXHfCTh")
wardData = ward.toObject()
wardData.system.specification.value = "8"

this.actor.createEmbeddedDocuments("Item", [wardData], {fromEffect : this.effect.id})
`
 Could we have a helper function like;

`let addedWard = game.whfr4e.AddToActor(this.actor, "Trait", "Ward")
addedWard .system.specification.value = "8" //would be better as a option input in the function
`

**Comment by Forien** [comment]
1. This is unsafe as you have no control over what specific item you want effect to add. What if there are 2 compendiums with that Trait? What if there is trait of the same name in the world? And they are all different? It's bad to assume it will just work.
2. moving function to utility and not passing the effect... how do you want to do the `fromEffect` part? You would need additional parameter
3. Your code would not update the Trait, so it would never set value of 8 to its specification. You would need to call `update()` on it

at that point there would be no much difference between these codes:

```js
// what you can do now differently
const wardData = (await fromUuid("Compendium.wfrp4e-core.items.Bvd2aZ0gQUXHfCTh")).toObject();
// this is trait from Core compendium and will always be the same item, so we know what it is
const ward = await this.actor.createEmbeddedDocuments("Item", [wardData], {fromEffect : this.effect.id});
ward?.update({"system.specification.value": 8});
```
```js
// your idea
const ward = await game.whfr4e.AddToActor(this.actor, "Trait", "Ward", {fromEffect: this});
// this is unknow item called "ward" and we don't know where it cames from. 
// Hell, could be undefined if function above didn't find it
ward?.update({"system.specification.value": 8}); 
```

**Comment by StevenEvans1966** [comment]
Apologies as a professional developer I tend to think in code, unfortunately my knowledge of the wfrp4e code base and javascript is at the "enough to be dangerous" level.   My post was to improve the transparency of the code and create a single point (function) of maintenance.   

this is the main problem:
 `let item= await fromUuid("Compendium.wfrp4e-core.items.Bvd2aZ0gQUXHfCTh")`

all I know when reading the script is that I is getting an Object from the core module.  Compare with this

`let item =  game.wfrp4e.FindItem("Core", "Trait", "Ward")`

I can now see I am looking in the Core module for the Trait Ward.   All we have to create a lookup to the UUid from the three keys.    I am assuming that items (Traits, spell etc.) have unique Names within a module.

**Comment by Forien** [comment]
Right, from the top of my head I know of a one instance where multiple items of the same `type` share a name
![image](https://github.com/moo-man/WFRP4e-FoundryVTT/assets/11304207/9a19ad55-2be8-457d-9ce0-d750b9a5fdf2)

It probably is not the case for most of other things. I know some traits and talents share a name, but they are different type, so it wouldn't matter.
